preprocessors/multistage-textbook-diagram.schema.json

```diff
@@ -0,0 +1,149 @@
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 1
+                }
+            }
+        }
+    },
+    "properties": {
+        "nodes": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "description": "A stage in the flow diagram",
+                "properties": {
+                    "segments": {
+                        "type": "array",
+                        "description": "Divides the entire stage into different segments where each segment indicates a noteworthy region or structure.",
+                        "items": {
+                            "type": "object",
+                            "description": "Contains information about each individual segment in the stage. Information such as area, centre of segment etc are included in this section",
+                            "properties": {
+                                "name": {
+                                    "description": "The name of the segment. Provides additional detail regarding subsections of a stage, if present.",
+                                    "type": "string"
+                                },
+                                "contours": {
+                                    "description": "The separate contours making up the segment.",
+                                    "type": "array",
+                                    "items": {
+                                        "$ref": "#/definitions/contour"
+                                    }
+                                },
+                                "centroid": {
+                                    "allOf": [
+                                        {
+                                            "$ref": "#/definitions/normCoord"
+                                        },
+                                        {
+                                            "description": "The overall centroid of the segment with normalized coordinates."
+                                        }
+                                    ]
+                                },
+                                "area": {
+                                    "description": "Total area occupied by the segment in a given stage. the area is normalised between 0 and 1",
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                }
+                            },
+                            "required": [
+                                "contours",
+                                "centroid",
+                                "area"
+                            ]
+                        }
+                    },
+                    "id": {
+                        "type": "string",
+                        "description": "Unique identifier for the stage"
```

Comment:
UUID?

Reply:
We are only using this to indicate the source and target in the 'links'.

Reply:
Yes but what is it?

Reply:
I am open to it being a number, a letter or a descriptive word (e.g. 'egg') - but if it is advisable to clearly define what this is/ limit to a single option, then numbers might be the best choice.